packages/server/lib/util/escape_filename.ts

```diff
@@ -0,0 +1,17 @@
+import _ from 'lodash'
+
+const ampersandRe = /&/g
+const percentRe = /%/g
+const questionRe = /\?/g
+
+export function escapeFilenameInUrl (url: string) {
+  let paths = url.split('/')
+
+  // escape valid file name characters that cannot be used in URL
+  paths[paths.length - 1] = (_.last<string>(paths) || '')
```

Comment:
This assumes only the file name might have these characters. If a folder has them (for example, `/cypress/integration/spec%?!/my_spec.js`), this won't work. We need to handle that and add a test for it too.

Reply:
Done.